rs/registry/canister/src/mutations/node_management/do_add_node.rs

```diff
@@ -76,20 +76,25 @@ impl Registry {
             ));
         }
 
-        // 3. Get valid type if type is in request
-        let node_reward_type = payload
-            .node_reward_type
-            .as_ref()
-            .map(|t| {
-                validate_str_as_node_reward_type(t).map_err(|e| {
-                    format!(
-                        "{}do_add_node: Error parsing node type from payload: {}",
-                        LOG_PREFIX, e
-                    )
+        // 3. Check if node reward type is valid, if node rewards table is not empty, and the field is in the request
+        // The node rewards table is empty on testnets and local deployments such as UTOPIA.
+        let node_reward_type = if self.is_node_rewards_table_empty() {
+            None
```

Comment:
@max-dfinity I had to add this check already since some tests were failing
```
Registration request failed: unexpected result: "rejected" - Some("Error from Canister rwlgt-iiaaa-aaaaa-aaaaa-cai: Canister called `ic0.trap` with message: [Registry]  Add node failed: [Registry] do_add_node: Error parsing node type from payload: Invalid node type: .\nConsider gracefully handling failures from this canister or altering the canister to handle exceptions. See documentation: http://internetcomputer.org/docs/current/references/execution-errors#trapped-explicitly")
```

Reply:
Hm - this doesn't make a lot of sense to me - the validation for the types doesn't use the rewards table, so it really shouldn't be related.  Are you sure it wasn't something else?  I think this would have the effect of hiding the error, but not really fixing it, in tests where you didn't also set up the NodeRewardsTable

Reply:
Agreed, let's deal with the validation in one of the follow-up PRs.